src/sif/motorcyclecost.cc

```diff
@@ -201,7 +201,19 @@ class MotorcycleCost : public DynamicCost {
    * @return  Returns true if access is allowed, false if not.
    */
   virtual bool Allowed(const baldr::NodeInfo* node) const {
-    return (node->access() & kMotorcycleAccess);
+    return (node->access() & (ignore_access_ ? kAllAccess : kMotorcycleAccess));
+  }
+
+  /**
+   * Checks if access is allowed for the provided edge. The access check based on mode
+   * of travel and the access modes allowed on the edge.
+   * @param   edge  Pointer to edge information.
+   * @return  Returns true if access is allowed, false if not.
+   */
+  virtual bool IsAccessable(const baldr::DirectedEdge* edge) const {
```

Comment:
we may need to inline these to avoid performance issues

Reply:
it might also be interesting to implement this only in DynamicCost and for each cost store the access type that applies to the underlying costing. honestly it would be great to move much of the common functionality up to the parent class if we can, but dont feel obligated to do that in the PR either. just something for future us to work on to make our lives easier!

Reply:
yeah, I also thought about it. I have some more ideas:
- looks like we can apply `ignore_access` flag in constructor of costing structure (`access_mask = ignore_access ? kAllAccess : kAutoAccess;`) and dont's touch this flag anymore

Reply:
> we may need to inline these to avoid performance issues

as far as I know, virtual methods can't be inlined in case we call base class. in this case, probably, it can be inlined because we call it from internal method